packages/plugins/transaction-logger/src/index.js

```diff
@@ -247,27 +245,44 @@ export default class TransactionLogger {
     this.embark.registerAPICall(
       'get',
       apiRoute,
-      (req, res) => {
-        res.send(JSON.stringify(this._getLogs()));
+      async (req, res) => {
+        res.send(JSON.stringify(await this._getLogs()));
       }
     );
   }
 
-  _getLogs() {
-    const data = this._readLogs();
+  async _getLogs() {
+    const data = await this._readLogs();
     return Object.values(data).reverse();
   }
 
   _saveLog(log) {
     this.writeLogFile.push(log);
   }
 
-  _readLogs() {
-    this.fs.ensureFileSync(this.logFile);
-    try {
-      return JSON.parse(this.fs.readFileSync(this.logFile));
-    } catch (_error) {
-      return {};
-    }
+  async _readLogs() {
+    return new Promise(async resolve => {
+      try {
+        await this.fs.ensureFile(this.logFile);
+        this.fs.readFile(this.logFile, (err, data) => {
+          if (err) {
+            throw (new Error(err));
```

Comment:
Shouldn't we `reject()` with an error here instead?

Reply:
No, the goal is that all errors are handled at the same place (line 282). That throw is intercepted and logged below